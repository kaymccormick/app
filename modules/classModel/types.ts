import { Map, List } from 'immutable';
import { ApplicationModule } from '../../src/ApplicationModule';
import {Pojo} from 'classModel';

export interface InputObject {
    [propName: string]: any;
}
export const REQUEST_PROJECTS = 'REQUEST_PROJECTS';
export const LOAD_PROJECTS= 'LOAD_PROJECTS';
export const RECEIVE_PROJECTS = 'RECEIVE_PROJECTS';
export const REQUEST_INITIAL_DATA = 'REQUEST_INITIAL_DATA';
export const FETCH_INITIAL_DATA = 'FETCH_INITIAL_DATA';
export const LOAD_INITIAL_DATA = 'LOAD_INITIAL_DATA';
export const RECEIVE_INITIAL_DATA = 'RECEIVE_INITIAL_DATA';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface Module extends ApplicationModule<ModuleState> {
}

export interface RequestProjectsAction {
    type: typeof REQUEST_PROJECTS;
}
export interface RequestInitialDataAction {
    type: typeof REQUEST_INITIAL_DATA;
}
export interface ReceiveInitialDataAction {
    type: typeof RECEIVE_INITIAL_DATA;
    result: InputObject;
}
export interface LoadInitialDataAction {
    type: typeof LOAD_INITIAL_DATA;
}
export interface FetchInitialDataAction {
    type: typeof FETCH_INITIAL_DATA;
}
export interface LoadProjectsAction {
    type: typeof LOAD_PROJECTS;

}
export interface ReceiveProjectsAction {
    type: typeof RECEIVE_PROJECTS;
    projects: List<Pojo.ProjectPojo>;
}

export type ActionTypes = ReceiveProjectsAction | LoadProjectsAction | RequestProjectsAction | RequestInitialDataAction |
ReceiveInitialDataAction | LoadInitialDataAction | FetchInitialDataAction;

export type EntitiesType = Map<string, Map<number, Pojo.BasePojo>> ;
export interface ModuleState {
/* Module state fields */
    entities: EntitiesType;
}
