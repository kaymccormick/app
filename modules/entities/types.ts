export interface ReceiveEntitiesAction {
    type: string;
    entities: EntityPojo[];
}
export interface EntitiesState {
    entities?: EntityPojo[];
}

export interface EntityPojo {
    name?: string;
    tableName?: string;
    columns?: EntityColumnPojo[];
}
export interface EntityColumnPojo {
    propertyName?: string;
    type?: string;
    isPrimary?: boolean;
    isNullable?: boolean;
    isGenerated?: boolean;
    comment?: string;
    isArray?: boolean ;
    propertyPath?: string;
    isObjectId?: boolean;
}
