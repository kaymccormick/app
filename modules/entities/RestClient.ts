import axios from 'axios';
import GetEntitiesResponse from './rest/response/GetEntitiesResponse';
export interface Args {
    baseUri: string;
}

export default class RestClient {
    public baseUri: string;

    public constructor(args: Args) {
        this.baseUri = args.baseUri;
    }
    public getEntities(): Promise<GetEntitiesResponse> {
        return axios.get(`${this.baseUri}/entity`).then(response => {
            if(response.status >= 400) {
                return new Error();
            }
            return response.data;
        }).then(data => {
            const r = new GetEntitiesResponse(data.success, data.result);
            return r;
        });
    }
}
