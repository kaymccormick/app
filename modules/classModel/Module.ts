import {ApplicationModule, ApplicationModuleArgs} from '../../src/ApplicationModule';
import {Configuration} from '../../src/Configuration';
import {Pojo} from 'classModel';
import reducer from './reducers';
import actions from './actions';
import { WebApplication } from '../../src/WebApplication';
import {ModuleState, EntitiesType} from './types';

import {Map,List,Set} from 'immutable';

import RestClient from '@heptet/rest-client'
/*export type Args = {
  restClient:RestClient,
}*/

export class Module extends ApplicationModule<ModuleState> {
    public restClient?: RestClient;
    public name: string = 'classModel';
    public key: string = 'classModel';

    public constructor(args: ApplicationModuleArgs) {
        super(args);
        this.actions = actions(args.restClient);
        this.restClient = args.restClient;
    }

    public getInitialState(): ModuleState {
        return {
            entities: Map<string, Map<number, Pojo.BasePojo>>(),
        }
    }
    

    public getReducers(): any {
        return reducer({restClient:this.restClient!});

    }
    public setup(app: WebApplication, config: Configuration) {
    }

    public getMainComponent(): Promise<any> {
        return import('./components/MainComponent');
    }
}


