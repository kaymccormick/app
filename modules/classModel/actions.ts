import {List,Map,Set} from 'immutable';
import { Pojo } from 'classModel';
import RestClient from '@heptet/rest-client';
import { EntitiesType, ActionTypes } from './types';
import { addMenuItems } from '../../modules/menus/actions';
import { MenuItemPojo } from '../../modules/menus/types';
import { Actions } from '../../src/types';

import {
    InputObject,
    RECEIVE_INITIAL_DATA,
    REQUEST_INITIAL_DATA,
    RECEIVE_PROJECT_DATA,
    REQUEST_PROJECT_DATA,
    } from './types';

export default (restClient: RestClient): Actions => {
    const receiveInitialData = (result: EntitiesType): ActionTypes =>{
        return { type: RECEIVE_INITIAL_DATA, result};
    }

    const receiveProjectData = (projects: List<Pojo.ProjectPojo>): ActionTypes =>{
        return { type: RECEIVE_PROJECT_DATA, projects };
    }

    const requestInitialData = (): ActionTypes => {
        return { type: REQUEST_INITIAL_DATA };
    }
    const requestProjectData = (): ActionTypes => {
        return { type: REQUEST_PROJECT_DATA };
    }

    const intermediateReceiveInitialData = (result: EntitiesType): (dispatch: any) => void => {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        return (dispatch: any): void => {
            let items = List<MenuItemPojo>();
            const projectsKey = 'projects';
            const projectItem = {title: 'Projects', parentKey: '', key: projectsKey, subItems: Set<string>() };
            result.get('Project').forEach((project: Pojo.ProjectPojo|undefined): void => {
                if(!project) {
                    return;
                }
                const childKey = `${projectsKey}-${project.id}`;
                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                const item = {title: project.name!,
		  parentKey: projectsKey, key: childKey,
		  };
		  projectItem.subItems = projectItem.subItems.add(childKey);
		  items = items.push(item);
            });
            items = items.push(projectItem);
	  dispatch(addMenuItems(items));
            /*            result.get('Module').forEach((module: Pojo.ModulePojo|undefined) => {
                if(!module) {
                    return;
                }
                const parentKey= `${projectsKey}-${module.projectId}`;
                const key = `${parentKey}-${module.id}`;
                const item = {title: module.name!, parentKey, key};
                dispatch(addMenuItem(item));
                dispatch(addMenuItem({title:'Classes', parentKey: key, key: `${key}-classes`}));
                dispatch(addMenuItem({title:'Interfaces', parentKey: key, key: `${key}-interfaces`}));
                dispatch(addMenuItem({title:'Imports', parentKey: key, key: `${key}-imports`}));
                dispatch(addMenuItem({title:'Exports', parentKey: key, key: `${key}-exports`}));
            });
            result.get('Class').forEach((class_: Pojo.ClassPojo|undefined) => {
                if(!class_) {
                    return;
                }
                const module_ = result.get('Module').get(class_.moduleId!) as Pojo.ModulePojo;
                if(!module_) {
                    return;
                }

                const parentKey= `${projectsKey}-${module_.projectId}-${module_.id}-classes`;
                const key = `${parentKey}-${class_.id}`;
                const item = {title: class_.name!, parentKey, key};
                dispatch(addMenuItem(item));
                dispatch(addMenuItem({title:'Methods', parentKey: key, key: `${key}-methods`}));
                dispatch(addMenuItem({title:'Properties', parentKey: key, key: `${key}-props`}));
            });
            
            result.get('Interface').forEach((interface_: Pojo.InterfacePojo|undefined) => {
                if(!interface_) {
                    return;
                }
                const module_ = result.get('Module').get(interface_.moduleId!) as Pojo.ModulePojo;
                if(!module_) {
                    return;
                }

                const parentKey= `${projectsKey}-${module_.projectId}-${module_.id}-interfaces`;
                const key = `${parentKey}-${interface_.id}`;
                const item = {title: interface_.name!, parentKey, key};
                dispatch(addMenuItem(item));
                dispatch(addMenuItem({title:'Methods', parentKey: key, key: `${key}-methods`}));
                dispatch(addMenuItem({title:'Properties', parentKey: key, key: `${key}-props`}));
            });
*/            
            
            dispatch(receiveInitialData(result));
        }
    }


    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const fetchInitialData = (): (dispatch: any) => Promise<any> => {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        return (dispatch: any): Promise<any> => {
            dispatch(requestInitialData());
	    return restClient.getAll().then((result: InputObject): Map<string, Map<number, Pojo.BasePojo>> => {
                let container = Map<string, Map<number, Pojo.BasePojo>>();
                Object.keys(result).forEach((key): void => {
                    const ary = result[key];
		    // how can we sanity check the value?
		    if(ary !== undefined && Array.isArray(ary)) {
		    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                        let map = Map<number, Pojo.BasePojo>(List<InputObject>(ary).filter((elem): boolean => elem !== undefined).map((elem: InputObject|undefined): [number, Pojo.BasePojo] => [elem!.id, elem!]));
                        container = container.set(key, map);
		    } else{
		    console.log(`invalid value ${ary} for key ${key} in result`)
		    }
		    });
		    return container;
		    }).then((container): void => {
                dispatch(intermediateReceiveInitialData(container));
		    });
        };
    };

    const fetchProjectData = (): (dispatch: any) => void => {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        return (dispatch: any): void => {
            dispatch(requestProjectData());
        }
    }

    return {
        requestInitialData,
        intermediateReceiveInitialData,
        fetchInitialData,
        receiveInitialData,
        receiveProjectData,
        fetchProjectData,
    };
};
