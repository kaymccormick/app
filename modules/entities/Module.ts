import uuidv4 from 'uuid/v4';
import {List,Map} from 'immutable';
import {ApplicationModule, ApplicationModuleArgs} from '../../src/ApplicationModule';
import {Configuration} from '../../src/Configuration';
import reducer from './reducers';
import * as actions from './actions';
import RestClient from './RestClient';
/*import EntitiesViewContainer from './components/containers/EntitiesViewContainer';*/
import { WebApplication } from '../../src/WebApplication';
import {EntitiesState,EntityPojo} from './types';

export class Module extends ApplicationModule<EntitiesState> {
    public restClient?: RestClient;
    public name: string = 'entities';
    public key: string = 'entities';
    public id: string;

    public constructor(args: ApplicationModuleArgs) {
        super(args);
        this.id = uuidv4();
        this.actions = actions;
//        console.log(`${this.id} setting rest client`);
        this.restClient = new RestClient({ baseUri: '/cme'});
//        console.log(`${this.id} ${this.restClient}`);
    }

    public getInitialState(): EntitiesState {
        return {
            entities: [],
                entitiesMap: Map<string,EntityPojo>(),
                    ui: { entities: [] },
        }
    }

    public getReducers(): any {
        if(!this.restClient) {
            throw new Error(`${this.id} No rest client`);
        }
        const r = reducer({module: this,restClient:this.restClient});
        console.log(r);
        return r;

    }
    public setup(app: WebApplication, config: Configuration) {
        //const restClient = config.getResource('restClient');
        //this.restClient = restClient;
    }
    public getMainComponent(): Promise<any> {
        return import('./components/MainComponent');
    }
/*    public getMenuTree(): Tree {
    }*/
}

        