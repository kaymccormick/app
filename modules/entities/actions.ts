import{List} from 'immutable';
import { Pojo } from 'classModel';
import RestClient from './RestClient';
import { EntityPojo, REQUEST_ENTITIES, LOAD_ENTITIES, RECEIVE_ENTITIES, ADD_SELECTED_ENTITIES, SELECT_ITEM, EntitiesActionTypes } from './types';
import GetEntitiesResponse from './rest/response/GetEntitiesResponse';
import { addMenuItem } from '../../modules/menus/actions';

export default () => {
const selectItem = (item: any): EntitiesActionTypes => {
    return { type: SELECT_ITEM, item };
};

const addSelectedEntities = (selectedEntities: any): EntitiesActionTypes => {
    return { type: ADD_SELECTED_ENTITIES, selectedEntities };
};

const requestEntities = (): EntitiesActionTypes => {
    return { type: REQUEST_ENTITIES };
}
const receiveEntities = (entities: EntityPojo[]): EntitiesActionTypes => {
    return { type: RECEIVE_ENTITIES, entities };
};
const intermediateReceiveEntities = (entities: EntityPojo[]): (dispatch: any) => void => {
return (dispatch: any) => {
  dispatch(receiveEntities(entities));
  dispatch(addMenuItem({title: 'Entities', parentKey: '', key: 'entities'}));
    entities.forEach((entity) => {
  });
  };
};

const fetchEntities = (restClient: RestClient) => {
    if(!restClient || !restClient.getEntities) {
        console.log(restClient);
        return { type: 'NOOP'};
    }
    return (dispatch: any) => {
        dispatch(requestEntities());
        return restClient.getEntities().then((response: GetEntitiesResponse) => {
            if(response.success) {
                if(response.result) {
                    dispatch(receiveEntities(response.result));
                }
            }
        });
    }

};

const loadEntities = (): EntitiesActionTypes => {
    console.log('load');
    return {type: LOAD_ENTITIES};
};

return { selectItem,
addSelectedEntities,
requestEntities,
receiveEntities,
intermediateReceiveEntities,
fetchEntities,
loadEntities,
};
};
