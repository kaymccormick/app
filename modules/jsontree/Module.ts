import uuidv4 from 'uuid/v4';
import {List,Map} from 'immutable';
import {ApplicationModule, ApplicationModuleArgs} from '../../src/ApplicationModule';
import {Configuration} from '../../src/Configuration';
import reducer from './reducers';
import * as actions from './actions';
import { WebApplication } from '../../src/WebApplication';
import {ModuleState} from './types';

export class Module extends ApplicationModule<ModuleState> {
    public name: string = 'jsontree';
    public key: string = 'jsontree';
    public id: string;

    public constructor(args: ApplicationModuleArgs) {
        super(args);
        this.id = uuidv4();
        this.actions = actions;
    }

    public getInitialState(): ModuleState {
        return {
        }
    }

    public getReducers(): any {
        return reducer({module: this});
    }
    public setup(app: WebApplication, config: Configuration) {
    }
    public getMainComponent(): Promise<any> {
        return import('./components/MainComponent');
    }
}

        