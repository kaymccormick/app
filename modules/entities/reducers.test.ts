import makeReducer from "./reducers";
import * as actions from "./actions";
import { EntityPojo } from "./types";
import { Module } from "./Module";
import { RestClient } from "./RestClient";
import { AppLogger } from '../../src/AppLogger';

test("1", () => {
  const mod = new Module({ logger: new AppLogger() });
  const restClient = new RestClient({ baseUri: "" });
  const reducer = makeReducer({ module: mod, restClient });
  const entities: EntityPojo[] = [{ name: "foo" }];
  // @ts-ignore
  const state = reducer(undefined, { type: "test" });
  expect(state).toMatchInlineSnapshot(`
    Object {
      "entities": Array [],
      "entitiesMap": Immutable.Map {},
      "ui": Object {
        "entities": Array [],
      },
    }
  `);
  const newState = reducer(state, actions.receiveEntities(entities));
  expect(newState).toMatchInlineSnapshot(`
    Object {
      "entities": Array [
        Object {
          "name": "foo",
        },
      ],
      "entitiesMap": Immutable.Map {
        "foo": Object {
          "name": "foo",
        },
      },
      "ui": Object {
        "entities": Array [],
      },
    }
  `);
});
