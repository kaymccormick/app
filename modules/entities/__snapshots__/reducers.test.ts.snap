// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state 1`] = `
Object {
  "entities": Array [],
  "entitiesMap": Immutable.Map {},
  "ui": Object {
    "entities": Array [],
  },
}
`;

exports[`state 2`] = `
Object {
  "entities": Array [
    Object {
      "name": "foo",
    },
  ],
  "entitiesMap": Immutable.Map {
    "foo": Object {
      "name": "foo",
    },
  },
  "ui": Object {
    "entities": Array [],
  },
}
`;
